### PR TITLE
chore: migrate core

### PR DESCRIPTION
**Pull request by flochtililoch**

### CODE REVIEW

File: src/core/components/hv-navigator/index.tsx [hunk 1]
```diff
@@ -10,9 +10,9 @@ import * as NavigationContext from 'hyperview/src/contexts/navigation';
 import * as NavigatorMapContext from 'hyperview/src/contexts/navigator-map';
 import * as NavigatorService from 'hyperview/src/services/navigator';
 import * as Types from './types';
-import * as TypesLegacy from 'hyperview/src/types-legacy';
+import * as TypesLegacy from 'hyperview/src/types';
```
Comment (hgray-instawork) [inline src/core/components/hv-navigator/index.tsx:13]: We'll want to remove the 'legacy' terminology after this merge. That was in place as a stopgap while developing new functionality in TS while the bulk was in JS.

File: src/core/components/modal/index.tsx [hunk 2]
```diff
@@ -24,7 +27,7 @@ import styles from './styles';
  * Uses styles defined on the <picker-field> element for the modal and buttons.
  * This is used on iOS only.
  */
-export default (props: Props): Node => {
+export default (props: Props): JSX.Element => {
```
Comment (hgray-instawork) [inline src/core/components/modal/index.tsx:30]: Not ReactNode?

File: src/core/hyper-ref/index.tsx [hunk 3]
```diff
@@ -361,14 +380,14 @@ export default class HyperRef extends PureComponent<Props, State> {
     );
   };
 
-  ScrollableView = ({ children }: { children: Node }): Node => {
+  ScrollableView = ({ children }: { children: JSX.Element }): JSX.Element => {
```
Comment (hgray-instawork) [inline src/core/hyper-ref/index.tsx:383]: vs. ReactNode

File: src/core/hyper-ref/index.tsx [hunk 4]
```diff
@@ -382,7 +401,7 @@ export default class HyperRef extends PureComponent<Props, State> {
     );
   };
 
-  VisibilityView = ({ children }: { children: Node }): Node => {
+  VisibilityView = ({ children }: { children: JSX.Element }): JSX.Element => {
```
Comment (hgray-instawork) [inline src/core/hyper-ref/index.tsx:404]: Same question

**Review by hgray-instawork — APPROVED** [review]

File: src/core/components/modal/index.tsx (hunk 2 above)
Comment (flochtililoch) [reply src/core/components/modal/index.tsx:30]: TSC was complaining about this, I can try revisit

File: src/core/components/hv-route/index.tsx [hunk 5]
```diff
@@ -169,11 +169,14 @@ class HvRouteInner extends PureComponent<Types.InnerRouteProps, Types.State> {
...
-        const styleSheet = Stylesheets.createStylesheets(preloadElement);
+        const styleSheet = Stylesheets.createStylesheets(
+          (preloadElement as unknown) as Document,
+        );
         const component:
           | string
           | React.ReactElement<unknown, string | JSXElementConstructor<unknown>>
-          | undefined = Render.renderElement(body, styleSheet, null, {
+          | undefined
+          | null = Render.renderElement(body, styleSheet, null, {
```
Comment (flochtililoch) [inline src/core/components/hv-route/index.tsx:179]: was the added `null` needed?

File: src/core/components/modal/index.tsx [hunk 6]
```diff
@@ -44,12 +47,13 @@ export default (props: Props): Node => {
     props.element.getAttribute('cancel-label') || 'Cancel';
   const doneLabel: string = props.element.getAttribute('done-label') || 'Done';
 
-  const getTextStyle = (pressed: boolean): Array<StyleSheetType> =>
-    createStyleProp(props.element, props.stylesheets, {
+  const getTextStyle = (pressed: boolean): Array<StyleSheetType> => {
+    return createStyleProp(props.element, props.stylesheets, {
```
Comment (flochtililoch) [inline src/core/components/modal/index.tsx:51]: revert

File: src/core/components/modal/index.tsx [hunk 7]
```diff
@@ -58,13 +62,13 @@ export default (props: Props): Node => {
...
 
-  const onLayout = (event: LayoutEvent) => {
+  const onLayout = (event: LayoutChangeEvent) => {
     setHeight(event.nativeEvent.layout.height);
   };
 
   const getDuration = (attribute: string, defaultValue: number) => {
-    const value = parseInt(props.element.getAttribute(attribute), 10);
-    return Number.isNaN(value) || value < 0 ? defaultValue : value;
+    const value = Dom.safeParseIntAttribute(props.element, attribute);
```
Comment (flochtililoch) [inline src/core/components/modal/index.tsx:70]: revert refactor

File: src/core/components/modal/index.tsx [hunk 8]
```diff
@@ -148,9 +151,11 @@ export default (props: Props): Node => {
               onPress={onDone}
             />
           </View>
-          {props.children}
+          <>{props.children}</>
```
Comment (flochtililoch) [inline src/core/components/modal/index.tsx:154]: check if still needed

File: src/core/hyper-ref/index.tsx [hunk 9]
```diff
@@ -52,8 +49,9 @@ import { createTestProps } from 'hyperview/src/services';
  */
 export const createEventHandler = (
   handler: () => void,
-  preventDefault: boolean = false,
-): ((event: any) => void) => event => {
+  preventDefault = false,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): ((event?: any) => void) => event => {
```
Comment (flochtililoch) [inline src/core/hyper-ref/index.tsx:54]: signature changed

File: src/core/hyper-ref/index.tsx [hunk 10]
```diff
@@ -198,8 +205,9 @@ export default class HyperRef extends PureComponent<Props, State> {
       };
     }
     // Custom behavior
-    return (element: Element) =>
-      onUpdate(null, action, element, { behaviorElement, custom: true });
+    return (element: Element) => {
+      return onUpdate(null, action, element, { behaviorElement, custom: true });
```
Comment (flochtililoch) [inline src/core/hyper-ref/index.tsx:209]: revert

File: src/core/hyper-ref/index.tsx [hunk 11]
```diff
@@ -239,13 +247,12 @@ export default class HyperRef extends PureComponent<Props, State> {
...
-    const behaviors = this.behaviorElements.filter(
-      e =>
-        PRESS_TRIGGERS.indexOf(
-          e.getAttribute(ATTRIBUTES.TRIGGER) || TRIGGERS.PRESS,
-        ) >= 0,
-    );
+  TouchableView = ({ children }: { children: JSX.Element }): JSX.Element => {
+    const behaviors = this.behaviorElements.filter(e => {
+      const trigger = e.getAttribute(ATTRIBUTES.TRIGGER) || TRIGGERS.PRESS;
+      return PRESS_TRIGGERS.indexOf(trigger as PressTrigger) >= 0;
```
Comment (flochtililoch) [inline src/core/hyper-ref/index.tsx:253]: revert refactor

File: src/core/hyper-ref/index.tsx [hunk 12]
```diff
@@ -255,28 +262,32 @@ export default class HyperRef extends PureComponent<Props, State> {
...
+            oldHandler();
+            setTimeout(() => handler(this.props.element), time);
+            time += 1;
+          },
+        );
       } else {
-        pressHandlers[triggerPropName] = createEventHandler(() =>
-          handler(this.props.element),
+        pressHandlers[triggerPropName as PressPropName] = createEventHandler(
+          () => {
```
Comment (flochtililoch) [inline src/core/hyper-ref/index.tsx:288]: revert

File: src/core/hyper-ref/index.tsx [hunk 13]
```diff
@@ -293,24 +304,28 @@ export default class HyperRef extends PureComponent<Props, State> {
       });
     }
 
+    const delayOnPress = () => {
```
Comment (flochtililoch) [inline src/core/hyper-ref/index.tsx:307]: not clear where this is coming from

File: src/core/hyper-ref/index.tsx [hunk 14]
```diff
@@ -391,9 +410,9 @@ export default class HyperRef extends PureComponent<Props, State> {
       // We don't want to use the cached `behaviors` list here because
       // the DOM might have been mutated since.
       this.getBehaviorElements(TRIGGERS.VISIBLE)
-        .map(behaviorElement =>
-          this.createActionHandler(behaviorElement, this.props.onUpdate),
-        )
+        .map(behaviorElement => {
+          return this.createActionHandler(behaviorElement, this.props.onUpdate);
```
Comment (flochtililoch) [inline src/core/hyper-ref/index.tsx:414]: revert

File: src/core/components/hv-route/index.tsx (hunk 5 above)
Comment (hgray-instawork) [reply src/core/components/hv-route/index.tsx:179]: I wouldn't have thought so

File: src/core/hyper-ref/index.tsx (hunk 9 above)
Comment (flochtililoch) [reply src/core/hyper-ref/index.tsx:54]: necessary

File: src/core/hyper-ref/index.tsx [hunk 15]
```diff
@@ -361,14 +380,14 @@ export default class HyperRef extends PureComponent<Props, State> {
...
+  ScrollableView = ({ children }: { children: JSX.Element }): JSX.Element => {
     const behaviors = this.getBehaviorElements(TRIGGERS.REFRESH);
     if (!behaviors.length) {
       return children;
     }
-    const refreshHandlers = behaviors.map(behaviorElement =>
-      this.createActionHandler(behaviorElement, this.props.onUpdate),
-    );
+    const refreshHandlers = behaviors.map(behaviorElement => {
+      return this.createActionHandler(behaviorElement, this.props.onUpdate);
```
Comment (flochtililoch) [inline src/core/hyper-ref/index.tsx:389]: revert

File: src/core/hyper-ref/index.tsx [hunk 16]
```diff
@@ -414,7 +432,9 @@ export default class HyperRef extends PureComponent<Props, State> {
       VisibilityDetectingView,
       {
         id,
-        onInvisible: null,
+        onInvisible: () => {
```
Comment (flochtililoch) [inline src/core/hyper-ref/index.tsx:435]: revert

File: src/core/components/modal/index.tsx (hunk 2 above)
Comment (flochtililoch) [reply src/core/components/modal/index.tsx:29]: Screenshot 2023-10-05 at 9 55 29 AM

**Pull request merged**